vortex-datafusion/src/lib.rs

```diff
@@ -446,6 +446,10 @@ impl Stream for VortexRecordBatchStream {
 
         Poll::Ready(Some(Ok(batch)))
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (self.num_chunks, Some(self.num_chunks))
```

Comment:
this doesn't have a noticeable impact, but might be helpful to some downstream consumer maybe